### ISSUE TITLE
bug: Gnome on X11 fails to log in with megasync enabled

### ISSUE THREAD

**Issue by arakan94** [nix-community/home-manager#5910]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

When I enable megasync in my home.nix using `services.megasync.enable = true;`, GNOME fails to log in with following error (and screen showing "Oh no! Something has gone wrong."):

```
zář 29 13:54:18 nixos-david gnome-shell[2222]: Window manager error: Another compositing manager is already running on screen 0 on display “:0”.
zář 29 13:54:18 nixos-david systemd[2024]: org.gnome.Shell@x11.service: Failed with result 'protocol'.
zář 29 13:54:18 nixos-david systemd[2024]: Failed to start GNOME Shell on X11.
zář 29 13:54:18 nixos-david systemd[2024]: org.gnome.Shell@x11.service: Triggering OnFailure= dependencies.
zář 29 13:54:18 nixos-david systemd[2024]: org.gnome.Shell@x11.service: Scheduled restart job, restart counter is at 1.
zář 29 13:54:18 nixos-david systemd[2024]: Started GNOME Session Failed lockdown screen (user).
zář 29 13:54:18 nixos-david systemd[2024]: Disable GNOME Shell extensions after failure was skipped because of an unmet condition check (ConditionPathExists=/run/user/1000/gnome-shell-disable-extensions).
zář 29 13:54:18 nixos-david systemd[2024]: Starting GNOME Shell on X11...
zář 29 13:54:18 nixos-david .gnome-shell-wr[2323]: Running GNOME Shell (using mutter 46.2) as a X11 window and compositing manager
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-0: disconnected
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-0: Internal TMDS
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-0: 330.0 MHz maximum pixel clock
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0):
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): Vestel 43UHD_LCD_TV (DFP-1): connected
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): Vestel 43UHD_LCD_TV (DFP-1): Internal TMDS
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): Vestel 43UHD_LCD_TV (DFP-1): 600.0 MHz maximum pixel clock
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0):
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-2: disconnected
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-2: Internal DisplayPort
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-2: 1440.0 MHz maximum pixel clock
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0):
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-3: disconnected
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-3: Internal TMDS
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0): DFP-3: 165.0 MHz maximum pixel clock
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (--) NVIDIA(GPU-0):
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (EE) libinput: Compx 2.4G Receiver Mouse: Failed to set scroll to twofinger
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (EE) libinput: Compx 2.4G Receiver Mouse: Failed to set scroll to twofinger
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (EE) libinput: Compx 2.4G Receiver Mouse: Failed to set scroll to twofinger
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (EE) libinput: Compx 2.4G Receiver Mouse: Failed to set scroll to twofinger
zář 29 13:54:18 nixos-david /nix/store/4wni6vqcj4cgn7qhlvypq8hbjivaj9dh-gdm-46.2/libexec/gdm-x-session[2109]: (EE) libinput: Compx 2.4G Receiver Mouse: Failed to set scroll to twofinger
zář 29 13:54:20 nixos-david gnome-shell[2323]: Window manager error: Another compositing manager is already running on screen 0 on display “:0”.
zář 29 13:54:20 nixos-david systemd[2024]: org.gnome.Shell@x11.service: Failed with result 'protocol'.
zář 29 13:54:20 nixos-david systemd[2024]: Failed to start GNOME Shell on X11.
```

I tried adding `xsession.enable = true;` but nothing changed.

I do have these relevant settings in my configuration.nix:
```
services.xserver.enable = true;
services.xserver.displayManager.gdm.enable = true;
services.xserver.desktopManager.gnome.enable = true;

  services.xserver.videoDrivers = ["nvidia"];
  hardware.nvidia = {
    modesetting.enable = true;
    powerManagement.enable = false;
    powerManagement.finegrained = false;
    open = false;
    nvidiaSettings = true;
    package = config.boot.kernelPackages.nvidiaPackages.stable;
  };
  hardware.nvidia-container-toolkit.enable = true
```

Gnome on Wayland works, but megasync crashes on login - seems to be this issue: #5889 
When ran manually after login, it works.

I also tried installing megasync in configuration.nix using `environment.systemPackages` and that works on both X11 and Wayland, though megasync isn't automatically starting. When I add it to Startup applications using GNOME Tweaks, it works.

### Maintainer CC

@GaetanLepage 

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 6.6.52, NixOS, 24.05 (Uakari), 24.05.20240927.f651414`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"home-manager-24.05.tar.gz, nixos-24.05"`
 - nixpkgs: `/nix/store/2vmkhm30kdnics1bnj9zgblh9yr4cpyw-source`
```

**Comment by knokelmaat** [comment]
You can download a snapshot of the fix in pull request #5902 here: 
https://github.com/knokelmaat/home-manager/tree/megasync-fix
Click on _Code_, then _Download ZIP_. Extract it.

I see you are using nixos. If you installed home-manager as a module and you use flakes (like me), you can switch to the custom home-manager using this command (change the path to where you extracted the zip):
`nixos-rebuild switch --override-input home-manager ~/Downloads/home-manager-megasync-fix`
If you are not using flakes, the command should be something like
`nixos-rebuild switch -I home-manager=$HOME/Downloads/home-manager-megasync-fix`

If you use a standalone home-manager installation, the commands are similar (see [home-manager contribution manual](https://nix-community.github.io/home-manager/index.xhtml#sec-contrib-getting-started)

It would be nice if you could test it and report if it also solves this issue, as I am unable to reproduce it using GNOME on X11.

**Comment by arakan94** [comment]
> It would be nice if you could test it and report if it also solves this issue, as I am unable to reproduce it using GNOME on X11.

Thanks, I can confirm this fixed it for me on both X11 and Wayland. Great job!

**Comment by GaetanLepage** [comment]
Fixed by #5902 (you may now close this issue).
